### PR TITLE
Modify model to support undo

### PR DESCRIPTION
**Pull request by rachx**
Pending test cases

### CODE REVIEW

**Review by burnflare — APPROVED** [review]

File: src/main/java/seedu/agendum/logic/commands/UndoCommand.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public UndoCommand() {}
+
+
+    @Override
+    public CommandResult execute() {
+        assert model != null;
+        try {
+            model.restorePreviousToDoList();
+            return new CommandResult(MESSAGE_SUCCESS);
+        } catch (EmptyStackException e) {
```
Comment (rachx) [inline src/main/java/seedu/agendum/logic/commands/UndoCommand.java:24]: Is there a need to create our own exception?

File: src/test/java/seedu/agendum/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -525,6 +525,51 @@ public void execute_find_matchesIfAnyKeywordPresent() throws Exception {
...
 
 
+    @Test
+    public void execute_undo_identifiesNoPreviousCommand() throws Exception {
+        assertCommandBehavior("undo", UndoCommand.MESSAGE_FAILURE, new ToDoList(), Collections.emptyList());
+    }
+
+    @Test
+    public void execute_undo_reversePreviousMutatingCommand() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
```
Comment (rachx) [inline src/test/java/seedu/agendum/logic/LogicManagerTest.java:535]: Should we separate into different tests?

Comment (burnflare) [reply src/test/java/seedu/agendum/logic/LogicManagerTest.java:535]: For now its fine, I think in the future we should separate the test in this file into their own tests for each command. This file is growing to be a little too big right now. Filed #45

File: src/main/java/seedu/agendum/logic/commands/UndoCommand.java (hunk 1 above)
Comment (burnflare) [reply src/main/java/seedu/agendum/logic/commands/UndoCommand.java:24]: Yes, I think it should throw something like a `UndoCommandNotFoundException`.

**Pull request closed without merging**